Sources/Identity/CustomerInfoManager.swift

```diff
@@ -458,6 +580,58 @@ private extension CustomerInfoManager {
                                                sandboxEnvironmentDetector: BundleSandboxEnvironmentDetector.default)
         return previewCustomerInfo
     }
+
+    private func trackGetCustomerInfoStartedIfNeeded(trackDiagnostics: Bool) {
+        if #available(iOS 15.0, tvOS 15.0, macOS 12.0, watchOS 8.0, *), trackDiagnostics {
+            self.diagnosticsTracker?.trackGetCustomerInfoStarted()
+        }
+    }
+
+    private func trackGetCustomerInfoResultIfNeeded(trackDiagnostics: Bool,
+                                                    startTime: Date,
+                                                    cacheFetchPolicy: CacheFetchPolicy,
+                                                    customerInfoDataResult: CustomerInfoDataResult) {
+        self.trackGetCustomerInfoResultIfNeeded(
+            trackDiagnostics: trackDiagnostics,
+            startTime: startTime,
+            cacheFetchPolicy: cacheFetchPolicy,
+            hadUnsyncedPurchasesBefore: customerInfoDataResult.hadUnsyncedPurchasesBefore,
+            usedOfflineEntitlements: customerInfoDataResult.usedOfflineEntitlements,
+            result: customerInfoDataResult.result.mapError({ $0 as Error })
+        )
+    }
+
+    // swiftlint:disable:next function_parameter_count
+    private func trackGetCustomerInfoResultIfNeeded(trackDiagnostics: Bool,
+                                                    startTime: Date,
+                                                    cacheFetchPolicy: CacheFetchPolicy,
+                                                    hadUnsyncedPurchasesBefore: Bool?,
+                                                    usedOfflineEntitlements: Bool,
+                                                    result: Swift.Result<CustomerInfo, Error>) {
+        if #available(iOS 15.0, tvOS 15.0, macOS 12.0, watchOS 8.0, *), trackDiagnostics {
+
+            let error: PurchasesError?
+            switch result.error {
+            case let purchasesError as PurchasesError:
+                error = purchasesError
+            case let purchasesErrorConvertible as PurchasesErrorConvertible:
+                error = purchasesErrorConvertible.asPurchasesError
+            case let otherError:
```

Comment:
Hmm I'm wondering, will it ever enter here? Seems like for any `Swift.Error`, it will enter the previous case if I understand correctly?

Reply:
Not exactly. Because not all error types implement the `PurchasesErrorConvertible` protocol. E.g. a `DecodingError` in `func cachedCustomerInfo(appUserID: String) throws -> CustomerInfo?` will fall into this `otherError` category, because `DecodingError` does not implement our `PurchasesErrorConvertible` protocol.

In any case, we need to deal with the `default` case in the `switch` to make it exhaustive.